### PR TITLE
MINIFICPP-1636 - Use std::filesystem instead of boost and platform specific implementations

### PR DESCRIPTION
**Pull request by rustammendel**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/include/utils/file/FileUtils.h [hunk 1]
```diff
@@ -25,14 +25,7 @@
...
 
-#ifdef USE_BOOST
-#include <dirent.h>
-#include <boost/filesystem.hpp>
-#include <boost/system/error_code.hpp>
-#ifndef WIN32
-#include <sys/stat.h>
-#endif
-#else
+#ifndef USE_BOOST
```
Comment (szaszm) [inline libminifi/include/utils/file/FileUtils.h:28]: We can now remove this check and just assume that it's not defined. We will eventually get rid of all boost code from libminifi.

File: docker/DockerVerify.sh [hunk 2]
```diff
@@ -44,11 +44,11 @@ echo "Installing test dependencies..." 1>&2
...
 if brew list 2> /dev/null | grep openssl > /dev/null 2>&1; then
   echo "Using homebrew paths for openssl" 1>&2
-  LDFLAGS="-L$(brew --prefix openssl)/lib"
+  LDFLAGS="-L$(brew --prefix openssl@1.1)/lib"
   export LDFLAGS
-  CFLAGS="-I$(brew --prefix openssl)/include"
+  CFLAGS="-I$(brew --prefix openssl@1.1)/include"
   export CFLAGS
-  SWIG_FEATURES="-cpperraswarn -includeall -I$(brew --prefix openssl)/include"
+  SWIG_FEATURES="-cpperraswarn -includeall -I$(brew --prefix openssl@1.1)/include"
```
Comment (szaszm) [inline docker/DockerVerify.sh:51]: Why is this version pin? Was something not working on mac?

**Review by szaszm — COMMENTED** [review]
Thank you for your contribution, this is an awesome and very welcome change. I only have 2 small comments, which is super rare in a 500 line diff, especially in a first contribution.

File: docker/DockerVerify.sh (hunk 2 above)
Comment (rustammendel) [reply docker/DockerVerify.sh:51]: Yes SWIG libs have incompatibility with openssl@3, it was failing on local.

File: libminifi/include/utils/file/FileUtils.h (hunk 1 above)
Comment (rustammendel) [reply libminifi/include/utils/file/FileUtils.h:28]: I have tried removing it but that failed windows build. I will try to optimize imports by moving relevant includes to ifdef WIN32 scope

**Review by szaszm — APPROVED** [review]

**Review by martinzink — APPROVED** [review]
Thanks for your contribution. This PR certainly makes FileUtils more readable than before. :+1:

File: extensions/sftp/tests/PutSFTPTests.cpp [hunk 3]
```diff
@@ -504,7 +504,9 @@ TEST_CASE_METHOD(PutSFTPTestsFixture, "PutSFTP set mtime", "[PutSFTP]") {
   testController.runSession(plan, true);
 
   testFile("nifi_test/tstFile1.ext", "content 1");
-  testModificationTime("nifi_test/tstFile1.ext", 3000000000LL);
+  using namespace std::chrono;  // NOLINT(build/namespaces)
```
Comment (lordgamez) [inline extensions/sftp/tests/PutSFTPTests.cpp:507]: nitpick: I would use `std::literals::chrono_literals` namespace here instead as we allow that in the linter and use the fully qualified namespaces for the type declaration.

**Review by lordgamez — APPROVED** [review]
LGTM, thanks for this change!

File: extensions/script/python/PythonCreator.h [hunk 4]
```diff
@@ -121,41 +122,34 @@ class PythonCreator : public minifi::core::CoreComponent {
...
-    const auto python_dir = script_directory.substr(basePath.length() + 1);
-    auto python_package = python_dir.substr(0, python_dir.find_last_of("/\\"));
-    utils::StringUtils::replaceAll(python_package, "/", ".");
-    utils::StringUtils::replaceAll(python_package, "\\", ".");
+    const auto python_package_path = std::filesystem::relative(pythonscript, basePath).parent_path();
+    std::vector<std::string> path_elements{python_package_path.begin(), python_package_path.end()};
+    std::string python_package = minifi::utils::StringUtils::join(".", path_elements);
     if (python_package.length() > 1 && python_package.at(0) == '.') {
       python_package = python_package.substr(1);
     }
```
Comment (fgerlits) [inline extensions/script/python/PythonCreator.h:138]: I don't think this is needed any longer, as none of the path elements can be empty, and `StringUtils::join()` is not going to put a separator at the front.

Comment (rustammendel) [reply extensions/script/python/PythonCreator.h:138]: image
on my local it does

Comment (fgerlits) [reply extensions/script/python/PythonCreator.h:138]: That is very strange; I get "h2o.h2o3.mojo".  There is probably a bug somewhere, but it was most likely already there before this change, so we don't have to fix it now.

**Review by adam-markovics — APPROVED** [review]

**Review by fgerlits — APPROVED** [review]

**Comment by szaszm** [comment]
I copied the name and email address from the last commit on your branch when merging, but GitHub seems to associate it with a different account of yours. I hope it's not a problem.

**Pull request closed without merging**